main.js

```diff
@@ -1,31 +1,104 @@
 let api_url = 'https://jsonplaceholder.typicode.com/users';
-fetch(api_url)
-.then(function(response){
-    return response.json();
-})
-.then(function(information){
+let myArray=[];
+console.log(myArray);
+let data;
+let result;
+function getData(){
+    return fetch(api_url)
+    .then((response) => {
+        myArray = response.json();
+        return myArray;
+    }); 
+}
+console.log(myArray)
+// second way
+// async function getDataMain(){
+//     let meta = await fetch(api_url);
+//     return meta.json()
+// }
+async function buildTable(result = []){
     let placeholder = document.querySelector("#data-output");
     let out="";
-    for(let info of information){
-        out+=`
-            <tr>
-            <td>${info.id}</td>
-            <td>${info.name}</td>
-            <td>${info.username}</td>
-            <td>${info.email}</td>
-            <td>${info.address.street},${info.address.suite},${info.address.city},${info.address.zipcode},${info.address.geo.lat},${info.address.geo.lng}</td>
-            <td>${info.phone}</td>
-            <td>${info.website}</td>
-            <td>${info.company.name},${info.company.catchPhrase},${info.company.bs}</td>
-            </tr>
-        
-        `;
+    placeholder.innerHTML = '';
+    if(result.length == 0){
+        let myArray = await getData();
+        for(let info of myArray){
+            out+=`
+                <tr>
+                <td>${info.id}</td>
```

Comment:
use object destructuring to get data from info object

Reply:
i will search for this topic ...